cmd/main.go

```diff
@@ -27,6 +33,16 @@ func main() {
 	fs := flag.NewFlagSet("aws-ebs-csi-driver", flag.ExitOnError)
 	options := GetOptions(fs)
 
+	cloud.RegisterMetrics()
+	mux := http.NewServeMux()
+	mux.Handle("/metrics", legacyregistry.HandlerWithReset())
+	go wait.Until(func() {
+		err := http.ListenAndServe("", mux)
+		if err != nil {
+			klog.Errorf("failed to listen & serve metrics: %v", err)
+		}
+	}, 5*time.Second, wait.NeverStop)
```

Comment:
Yeah let me add that, I think if i try to listen on port 80 with hostnetwork true it could conflict with other stuff.

basically the wait.util is to loop forever and retry every 5s in case the server stops for whatever reason (for example, if the port is taken by something else? (i copied this pattern from somewhere else in k8s code base...)

Alternatively we can panic/klog.Fatal and restart the whole process in case the server stops for whatever reason. I think i'll change it to do that, otherwise users will be confused if the pod appears healthy but there r no metrics.

Reply:
added --http-endpoint (name consistent with sidecars) and klog.Fatal on error.